Create merge_queue.md (#158959)
flutter/flutter: docs/infra/merge_queue.md

## Github Merge Queue

If you are not hacking on flutter/flutter, you can stop reading now.

tl;dr GitHub merge queue will be enabled in the flutter/flutter repo, which slightly changes the PR merge process, but you should not notice a change in your workflow.

> On Friday November 14, 2024, GitHub merge queue feature will be enabled in the flutter/flutter repo. After this change, pull requests will first need to pass merge queue tests before landing on the target branch (e.g. main or master). At this time, no additional tests will run in the merge queue. A PR that passes presubmit checks will be allowed to land on the target branch immediately.

Reviewer: It was pointed out to me that Friday is November 15 this year :)

Author: TIL. 😜

> On Friday November 14, 2024, GitHub merge queue feature will be enabled in the flutter/flutter repo. After this change, pull requests will first need to pass merge queue tests before landing on the target branch (e.g. main or master). At this time, no additional tests will run in the merge queue. A PR that passes presubmit checks will be allowed to land on the target branch immediately.

Author: ```suggestion
On Friday November 15, 2024, GitHub merge queue feature will be enabled in the flutter/flutter repo. After this change, pull requests will first need to pass merge queue tests before landing on the target branch (e.g. main or master). At this time, no additional tests will run in the merge queue. A PR that passes presubmit checks will be allowed to land on the target branch immediately.
```

What should I expect?
The only visible change will be the merge button. Currently, the button looks like this:

> <!-- TODO(Piinks) Image -->

Author: ```suggestion
![Squash and merge button](https://github.com/flutter/assets-for-api-docs/blob/main/assets/wiki/squash_merge.png)
```

With the merge queue, the button will now look like this:

> <!-- TODO(Piinks) Image -->

Author: ```suggestion
![Merge when ready button](https://github.com/flutter/assets-for-api-docs/blob/main/assets/wiki/merge_ready.png)
```

Unlike "Squash and merge", the new button does not immediately land the PR on the target branch, but instead enqueues your PR for extra checks in the merge queue. If you do not do anything, eventually your PR will land.
Can I still use autosubmit?
Yes! Feel free to continue using the autosubmit label as normal. The bots will run your PR through the merge queue and land it automatically.
What's "Remove from queue"? Can I click it?
While your PR is enqueued, Github will show the following button:

> <!-- TODO(Piinks) Image -->

Author: ```suggestion
![Remove from queue button](https://github.com/flutter/assets-for-api-docs/blob/main/assets/wiki/remove_queue.png)
```

When you press this button, GitHub will remove your PR from the queue, and the PR will no longer be landed by any automated machinery.

WARNING: please only use this button in case of emergency. Adding and removing your PR from the queue costs in CI resources. Any CI work being done for your PR and all PRs that follow yours will be stopped and restarted.
What if I have questions?

> If you have any questions or concerns, or If you suspect you are unable to land your PRs for reasons other than the familiar test failures and flakes, please contact any of: yjbanov@google.com, codefu@google.com, fujino@google.com, katelovett@google.com.

Author: ```suggestion
If you have any questions or concerns, or If you suspect you are unable to land your PRs for reasons other than the familiar test failures and flakes, please contact any of: @yjbanov, @jtmcdole, @christopherfujino, @Piinks.
```
